app/workers/go_project_verification_worker.rb

```diff
@@ -14,8 +14,17 @@ def perform(name)
     # check to see if the module is found on pkg.go.dev and if it isn't then go ahead and delete this name
     return project.destroy unless PackageManager::Go.valid_project?(project.name)
 
-    # if this name is found and is considered a module then that should be considered canonical
-    unless PackageManager::Go.module?(project.name)
+    # if this name is found and is considered a module then we can determine the canonical name
+    if PackageManager::Go.module?(project.name)
+      canonical_name = PackageManager::Go.canonical_module_name(project.name)
```

Comment:
if `module?` returns true, doesn't that mean that we have the canonical name by this point? For instance, this would return `false` because the HTML says it doesn't have a valid go.mod: https://pkg.go.dev/github.com/go-sql-driver/mysQL

Reply:
Going forward yes, but we have a bunch of duplicates for names that should be canonical right now. This is updating the worker to remove the non canonical name since we for sure know it's incorrect now.